parse_event_logs.py

```diff
@@ -180,6 +180,33 @@ def output_utilizations(self, prefix):
     self.__write_utilization_summary_file(
       process_system_cpu_utilizations, "%s_%s" % (prefix, "cpu_process_system_utilization"))
 
+  def output_stage_resource_metrics(self, filename):
+    """ Writes a file with resource metrics at the granularity of stages. """
+    executor_id_to_host = self.get_executor_id_to_host()
+    with open("{}_{}".format(filename, "stage_resource_metrics"), "w") as output:
+      for job_id, job in sorted(self.jobs.iteritems()):
+        for stage_id, stage in sorted(job.stages.iteritems()):
+          executor_to_resource_metrics = stage.get_executor_id_to_resource_metrics()
+          for executor_id, resource_metrics in sorted(executor_to_resource_metrics.iteritems()):
+            output.write("Job {}, Stage {}, Executor {} ({}):\n{}\n".format(
+              job_id, stage_id, executor_id, executor_id_to_host[executor_id], resource_metrics))
+
+  def output_job_resource_metrics(self, filename):
+    """ Writes a file with resource metrics at the granularity of jobs. """
```

Comment:
ditto to the above, but with "stage" instead of "job" (the problem with the existing comments is it's not clear what "at the granularity of" means -- does this mean a new file for each job? A CDF over jobs?)